components/cam/src/dynamics/se/dp_coupling.F90

```diff
@@ -1,279 +1,280 @@
-    use gravity_waves_sources, only: gws_src_fnct
-    use dyn_comp,       only: frontgf_idx, frontga_idx, hvcoord
-    use phys_control,   only: use_gw_front
-    implicit none
-!-----------------------------------------------------------------------
-! !INPUT PARAMETERS:
-!
-    type(dyn_export_t), intent(inout) :: dyn_out    ! dynamics export 
-    type(physics_buffer_desc), pointer :: pbuf2d(:,:)
-! !OUTPUT PARAMETERS:
+    use physics_buffer,          only: physics_buffer_desc, pbuf_get_chunk, pbuf_get_field
+    use shr_vmath_mod,           only: shr_vmath_exp
+    use time_manager,            only: is_first_step
+    use viscosity_mod,           only: compute_zeta_C0
+    use cam_abortutils,          only: endrun
+    use gravity_waves_sources,   only: gws_src_fnct
+    use dyn_comp,                only: frontgf_idx, frontga_idx, hvcoord
+    use phys_control,            only: use_gw_front
+    use fv_physics_coupling_mod, only: dyn_to_fv_phys
 
+    implicit none
+    !---------------------------------------------------------------------------
+    ! INPUT PARAMETERS:
+    type(dyn_export_t), intent(inout)  :: dyn_out         ! dynamics export 
+    type(physics_buffer_desc), pointer :: pbuf2d(:,:)     ! physics buffer
+    ! OUTPUT PARAMETERS:
     type(physics_state), intent(inout), dimension(begchunk:endchunk) :: phys_state
-    type(physics_tend ), intent(inout), dimension(begchunk:endchunk) :: phys_tend
-    
-
-! LOCAL VARIABLES
-    type(element_t), pointer :: elem(:)               ! pointer to dyn_out element array
-    integer (kind=int_kind)  :: ie               ! indices over elements
-    integer (kind=int_kind)  :: lchnk, icol, ilyr      ! indices over chunks, columns, layers
-    real (kind=real_kind)    :: ps_tmp(npsq,nelemd)         ! temporary array to hold ps
-    real (kind=real_kind)    :: phis_tmp(npsq,nelemd)       ! temporary array to hold phis  
-    real (kind=real_kind)    :: T_tmp(npsq,pver,nelemd)     ! temporary array to hold T
-    real (kind=real_kind)    :: uv_tmp(npsq,2,pver,nelemd)     ! temporary array to hold u and v
-    real (kind=real_kind)    :: q_tmp(npsq,pver,pcnst,nelemd) ! temporary to hold advected constituents
-    real (kind=real_kind)    :: omega_tmp(npsq,pver,nelemd) ! temporary array to hold omega
-
+    type(physics_tend ), intent(inout), dimension(begchunk:endchunk) :: phys_tend 
+    ! LOCAL VARIABLES
+    real(kind=real_kind), dimension(npsq,nelemd)            :: ps_tmp ! temp array to hold ps
+    real(kind=real_kind), dimension(npsq,nelemd)            :: zs_tmp ! temp array to hold phis  
+    real(kind=real_kind), dimension(npsq,pver,nelemd)       :: T_tmp  ! temp array to hold T
+    real(kind=real_kind), dimension(npsq,2,pver,nelemd)     :: uv_tmp ! temp array to hold u and v
+    real(kind=real_kind), dimension(npsq,pver,pcnst,nelemd) :: q_tmp  ! temp to hold advected constituents
+    real(kind=real_kind), dimension(npsq,pver,nelemd)       :: w_tmp  ! temp array to hold omega
```

Comment:
Omega is not the same as w. Maybe, confusion comes from the fact that omega=Dp/Dt and in pressure coordinates  is 'like' vertical velocity, but it is not. We reserve w for Dz/Dt.

Reply:
yes, I understand this, but as part of my effort to make this code readable this made sense. There will only be one variable that describes vertical velocity, and I expect any of the developers of this code to realize that "w" refers to vertical velocity, whether its relative to prssure or height. I doubt this will lead to any confusion, and its certainly not worse than the confusion surrounding whether an "omega" variable is actually "omega" or "omega*p", which is not always easy to determine from the code.

Reply:
We already use w in the dycore, both hydro and nonhydro. In nonhydro dycore it is a prognostic variable, and omega stays diagnostic. It is better to keep these two separated.

Reply:
What if I just change "w_tmp" to "o_tmp" or "om_tmp"?

Reply:
If om stands for omega, why not use omega?

Reply:
Just saw that you changed it to om_tmp.

Reply:
I just want to keep the variable names to a similar character length to make it easier to line things up for readability. Personally, I have a really hard time scanning through the code when things aren't lined up nicely, but it seems that most of the other developers don't mind this sort of thing. I can use omega if you think it would be better, but if no one cares then I'd prefer to keep it short.